src/unity/toolkits/object_detection/object_detector.cpp

```diff
@@ -326,16 +340,21 @@ std::shared_ptr<MLModelWrapper> object_detector::export_to_coreml(
 
   // Generate "user-defined" metadata.
   flex_dict user_defined_metadata = {
-    {"model", read_state<flex_string>("model")},
-    {"max_iterations", read_state<flex_int>("max_iterations")},
-    {"training_iterations", read_state<flex_int>("training_iterations")},
-    {"include_non_maximum_suppression", "False"},
-    // TODO: {"confidence_threshold", ... },
-    // TODO: {"iou_threshold", ...},
-    {"feature", read_state<flex_string>("feature")},
-    {"annotations", read_state<flex_string>("annotations")},
-    {"classes", class_labels_str},
-  };
+      {"model", read_state<flex_string>("model")},
+      {"max_iterations", read_state<flex_int>("max_iterations")},
+      {"training_iterations", read_state<flex_int>("training_iterations")},
+      {"include_non_maximum_suppression", "False"},
+      {"feature", read_state<flex_string>("feature")},
+      {"annotations", read_state<flex_string>("annotations")},
+      {"classes", class_labels_str},
+      {"type", "object_detector"},
+    };
+
+  if (options["include_non_maximum_suppression"] == "True") {
+    user_defined_metadata.push_back(std::make_pair<std::string, std::string>("include_non_maximum_suppression", options["include_non_maximum_suppression"]));
```

Comment:
You should be able to streamline this as e.g.
`user_defined_metadata.emplace_back("include_non_maximum_suppression", options["include_non_maximum_suppression"])`